src/main/java/com/teamacronymcoders/base/materialsystem/blocks/SubBlockOreSamplePart.java

```diff
@@ -44,13 +46,20 @@
     public SubBlockOreSamplePart(MaterialPart materialPart, MaterialUser materialUser) {
         super(materialPart, MaterialSystem.materialCreativeTab);
         MaterialPartData data = materialPart.getData();
-        this.mod = materialUser.getMod();
-        if (data.containsDataPiece(DROP_DATA_NAME)) {
-            itemDrop = data.getDataPiece(DROP_DATA_NAME);
-        }
-        if (data.containsDataPiece(ACTIVATED_TEXT_DATA_NAME)) {
-            activatedText = data.getDataPiece(ACTIVATED_TEXT_DATA_NAME);
-        }
+
+        setHardness(data.getValue("hardness", 0.125F, DataPartParsers::getFloat));
+        setResistance(data.getValue("resistance", 2, DataPartParsers::getInt));
+        setHarvestLevel(data.getValue("harvestLevel", -1, DataPartParsers::getInt));
+        setHarvestTool(data.getValue("harvestTool", null, DataPartParsers::getString));
+
+        mod = materialUser.getMod();

```

Comment:
Doesn't the super constructor take care of this?

Reply:
`SubBlockPart` does not have mod, `SubBlockOrePart` does, but I don't extend that.